src/resources/Resource.js

```diff
@@ -9,10 +10,23 @@ export default class Resource {
     this.api = api
   }
 
-  get apiUrl () {
-    const protocol = this.api.https ? 'https' : 'http'
-    const url = `${protocol}://${this.api.host}/userapi/v${this.api.version}`
-    return url
+  getApiUrl (source) {
```

Comment:
Does this need to be a function? I reckon it could still be the `getter` `get apiUrl()` because all the relevant information is retained in `this.api`.

If the `Availability` resource class is initialised with `this.source = foo`,
 this method can retrieve it.